tests/api/test_db.py

```diff
@@ -0,0 +1,20 @@
+from swoop.api.config import get_settings
+from swoop.api.db import close_db_connection, connect_to_db
+import pytest
+
+@pytest.mark.asyncio
+async def test_db_connection_pool():
+    settings = get_settings('.env')
+
+    pools = await connect_to_db(settings)
+    readpool = pools[0]
+
+    for i in range(0, settings.db_max_conn_size):
+        await readpool.acquire()
+
+    # No connections available, this should Timeout after 1s
+    with pytest.raises(TimeoutError):
+        await readpool.acquire(timeout=1)
+
+    readpool.close()
+    assert True
```

Comment:
I am assuming you are using vscode? Can you set [`"files.insertFinalNewline": true,`](https://github.com/microsoft/vscode/issues/12076)?